test/scalar/test_broadcast.jl

```diff
@@ -7,17 +7,13 @@ dv1 = DataValue(3)
 dv2 = DataValue(5)
 dv3 = DataValue{Int}()
 
-@test log.(dv1) .+ dv2 == DataValue(log(3) + 5)
-@test log.(dv1) .+ dv2 .- dv3 == DataValue{Float64}()
+@test log.(Ref(dv1)) .+ Ref(dv2) == DataValue(log(3) + 5)
```

Comment:
I can take out the `Ref` changes if you want; I think the deprecation warnings come from the fact that a `DataValue` is iterable? Is that important to DataValue? Otherwise, it might be easier w/ all the new broadcast machinery to leave a DataValue scalar as a proper scalar (no iterating).